ccan/ccan/graphql/graphql.h

```diff
@@ -0,0 +1,230 @@
+};
+
+struct graphql_string_value {
+	struct graphql_token *val;
+};
+
+struct graphql_boolean_value {
+	struct graphql_token *val;
+};
+
+struct graphql_float_value {
+	struct graphql_token *val;
+};
+
+struct graphql_int_value {
+	struct graphql_token *val;
+};
+
+struct graphql_value {
+	struct graphql_variable *var;
+	struct graphql_int_value *int_val;
+	struct graphql_float_value *float_val;
+	struct graphql_boolean_value *bool_val;
+	struct graphql_string_value *str_val;
+	struct graphql_null_value *null_val;
+	struct graphql_enum_value *enum_val;
+	struct graphql_list_value *list_val;
+	struct graphql_object_value *obj_val;
+};
+
+struct graphql_inline_fragment {
+	struct graphql_type_condition *type_cond;
+	struct graphql_directives *directives;
+	struct graphql_selection_set *sel_set;
+};
+
+struct graphql_type_condition {
+	struct graphql_named_type *named_type;
+};
+
+struct graphql_fragment_name {
+	struct graphql_token *name;
+};
+
+struct graphql_fragment_definition {
+	struct graphql_fragment_name *name;
+	struct graphql_type_condition *type_cond;
+	struct graphql_directives *directives;
+	struct graphql_selection_set *sel_set;
+};
+
+struct graphql_fragment_spread {
+	struct graphql_fragment_name *name;
+	struct graphql_directives *directives;
+};
+
+struct graphql_alias {
+	struct graphql_token *name;
+};
+
+struct graphql_argument {
+	struct graphql_argument *next;
+	struct graphql_token *name;
+	struct graphql_value *val;
+};
+
+struct graphql_arguments {
+	struct graphql_argument *first;
+};
+
+struct graphql_field {
+	struct graphql_alias *alias;
+	struct graphql_token *name;
+	struct graphql_arguments *args;
+	struct graphql_directives *directives;
+	struct graphql_selection_set *sel_set;
+};
+
+struct graphql_selection {
+	struct graphql_selection *next;
+	struct graphql_field *field;
+	struct graphql_fragment_spread *frag_spread;
+	struct graphql_inline_fragment *inline_frag;
+};
+
+struct graphql_selection_set {
+	struct graphql_selection *first;
+};
+
+struct graphql_operation_type {
+	struct graphql_token *op_type;
+};
+
+struct graphql_operation_definition {
+	struct graphql_operation_type *op_type;
+	struct graphql_token *op_name;
+	struct graphql_variable_definitions *vars;
+	struct graphql_directives *directives;
+	struct graphql_selection_set *sel_set;
+};
+
+struct graphql_executable_definition {
+	struct graphql_executable_definition *next_def;
+	struct graphql_operation_definition *op_def;
+	struct graphql_fragment_definition *frag_def;
+};
+
+struct graphql_executable_document {
+	struct graphql_executable_definition *first_def;
+};
+
+struct graphql_definition {
+	struct graphql_definition *next_def;
+	struct graphql_executable_definition *executable_def;
+	struct graphql_type_system_definition_or_extension *type_system_def;
+};
+
+struct graphql_document {
+	struct graphql_definition *first_def;
+};
+
+
+struct graphql_token {
+	struct list_node list;
+	unsigned int token_type;
+	unsigned int token_specific;
```

Comment:
It's unclear to me how this field is used?  It seem to be equal to token_type, or unset altogether?

Reply:
Deleted - it was just accidentally leftover from the development phase.